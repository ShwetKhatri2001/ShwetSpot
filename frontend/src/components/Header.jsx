import React from 'react';
import { Route, withRouter } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import { LinkContainer } from 'react-router-bootstrap';
import { Nav, Navbar, NavbarBrand, NavDropdown } from 'react-bootstrap';
import { logout } from '../redux/actions/userActions';
import SearchBox from './SearchBox';

const Header = () => {
  const dispatch = useDispatch();

  const userLogin = useSelector(state => state.userLogin);
  const { userInfo } = userLogin;

  const userDetails = useSelector(state => state.userDetails);
  const { user } = userDetails;

  const logoutHandler = () => {
    dispatch(logout());
  };

  return (
    <header>
      <Navbar expand="xl" collapseOnSelect id="main-header">
        <Navbar.Toggle aria-controls="basic-navbar-nav" />
        <Navbar.Collapse id="basic-navbar-nav">
          <Nav className="mr-auto" id="user-info">
            {userInfo && user ? (
              <>
                <NavDropdown title={`Hi ${userInfo.name}!`} id="username">
                  <LinkContainer to="/profile">
                    <NavDropdown.Item>Profile</NavDropdown.Item>
                  </LinkContainer>
                  <LinkContainer to="/productlist">
                    <NavDropdown.Item>Sell</NavDropdown.Item>
                  </LinkContainer>
                  <NavDropdown.Item onClick={logoutHandler}>
                    Logout
                  </NavDropdown.Item>
                </NavDropdown>
              </>
            ) : (
              <>
                <LinkContainer to="/login">
                  <Nav.Link id="sign-in-menu">
                    <i className="fas fa-user" /> Sign In
                  </Nav.Link>
                </LinkContainer>
              </>
            )}
            {userInfo && userInfo.isAdmin && (
              <NavDropdown title="Admin" id="admin-menu">
                <LinkContainer to="/admin/userlist">
                  <NavDropdown.Item>Users</NavDropdown.Item>
                </LinkContainer>
                <LinkContainer to="/admin/orderlist">
                  <NavDropdown.Item>Orders</NavDropdown.Item>
                </LinkContainer>
              </NavDropdown>
            )}
          </Nav>
          <Nav>
            <LinkContainer to="/cart" id="cart">
              <Nav.Link>
                <i className="fas fa-shopping-cart" /> Cart
              </Nav.Link>
            </LinkContainer>
          </Nav>
        </Navbar.Collapse>
      </Navbar>
      <Nav>
        <LinkContainer to="/">
          <NavbarBrand id="shop-brand">ShwetSpot</NavbarBrand>
        </LinkContainer>

        <Route render={({ history }) => <SearchBox history={history} />} />
      </Nav>
    </header>
  );
};

export default withRouter(Header);
