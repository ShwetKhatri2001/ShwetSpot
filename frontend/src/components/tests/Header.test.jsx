import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';

import Header from '../Header';
import store from '../../redux/store';

describe('Header Component', () => {
  const MockHeader = (
    <MemoryRouter>
      <Provider store={store}>
        <Header />
      </Provider>
    </MemoryRouter>
  );

  it('should render correctly', () => {
    const wrapper = shallow(MockHeader);
    expect(wrapper.contains(<Header />)).toBe(true);
  });

  it('should match the snapshot', () => {
    const tree = renderer.create(MockHeader).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
