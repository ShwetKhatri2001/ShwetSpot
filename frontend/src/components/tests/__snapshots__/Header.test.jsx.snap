// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should match the snapshot 1`] = `
<header>
  <nav
    className="navbar navbar-expand-xl navbar-light"
    id="main-header"
  >
    <button
      aria-controls="basic-navbar-nav"
      aria-label="Toggle navigation"
      className="navbar-toggler collapsed"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      aria-expanded={null}
      className="navbar-collapse collapse"
      id="basic-navbar-nav"
    >
      <div
        className="mr-auto navbar-nav"
        id="user-info"
        onKeyDown={[Function]}
      >
        <a
          className="nav-link"
          data-rb-event-key="/login"
          href="/login"
          id="sign-in-menu"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          <i
            className="fas fa-user"
          />
           Sign In
        </a>
      </div>
      <div
        className="navbar-nav"
        onKeyDown={[Function]}
      >
        <a
          className="nav-link"
          data-rb-event-key="/cart"
          href="/cart"
          id="cart"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          <i
            className="fas fa-shopping-cart"
          />
           Cart
        </a>
      </div>
    </div>
  </nav>
  <div
    className="nav"
    onKeyDown={[Function]}
  >
    <a
      className="active navbar-brand"
      href="/"
      id="shop-brand"
      onClick={[Function]}
      style={Object {}}
    >
      greenBay
    </a>
    <form
      className="mx-auto d-inline-flex"
      onSubmit={[Function]}
    >
      <input
        className="mt-3 form-control"
        id="search-box"
        name="q"
        onChange={[Function]}
        placeholder="Search Products..."
        type="text"
      />
      <button
        className="btn btn-success"
        disabled={false}
        id="search-box-btn"
        type="submit"
      >
        Search
      </button>
    </form>
  </div>
</header>
`;
